src/commands/ext-dev-deprecate.ts

```diff
@@ -0,0 +1,77 @@
+import * as clc from "cli-color";
+
+import { Command } from "../command";
+import { promptOnce } from "../prompt";
+import { ensureExtensionsApiEnabled, logPrefix } from "../extensions/extensionsHelper";
+import { deprecateExtensionVersion, listExtensionVersions } from "../extensions/extensionsApi";
+import * as refs from "../extensions/refs";
+import { parseVersionPredicate, compareVersions } from "../extensions/versionHelper";
+import { requireAuth } from "../requireAuth";
+import { FirebaseError } from "../error";
+import * as utils from "../utils";
+
+/**
+ * Deprecate all extension versions that match the version predicate.
+ */
+export default new Command("ext:dev:deprecate <extensionRef> <versionPredicate>")
+  .description("deprecate extension versions that match the version predicate")
+  .option("-m, --message <deprecationMessage>", "deprecation message")
+  .option(
+    "-f, --force",
+    "override deprecation message for existing deprecated extension versions that match"
+  )
+  .before(requireAuth)
+  .before(ensureExtensionsApiEnabled)
+  .action(async (extensionRef: string, versionPredicate: string, options: any) => {
+    const { publisherId, extensionId, version } = refs.parse(extensionRef);
+    const baseExtensionRef = publisherId + "/" + extensionId;
```

Comment:
This seems unnecessary - you immediately error out if the provided extensionRef isn't in this format, so you can probably just use that.